packages/p2p/src/components/modal-manager/modals/create-ad-add-payment-method-modal/create-ad-add-payment-method-modal.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Button, DesktopWrapper, Icon, MobileFullPageModal, MobileWrapper, Modal } from '@deriv/components';
+import { observer } from '@deriv/stores';
+import { localize, Localize } from 'Components/i18next';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import AddPaymentMethod from 'Pages/my-profile/payment-methods/add-payment-method/add-payment-method';
+import { useStores } from 'Stores';
+
+const CreateAdAddPaymentMethodModal = () => {
+    const { hideModal, is_modal_open, showModal } = useModalManagerContext();
+    const { general_store, my_profile_store } = useStores();
+    const { is_form_modified } = general_store;
+    const { selected_payment_method } = my_profile_store;
+
+    const onCancel = () => {
+        if (selected_payment_method.length > 0 || is_form_modified) {
+            showModal({
+                key: 'CancelAddPaymentMethodModal',
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
I'm a little concerned about this.
Since this is the first component that uses CancelAddPaymentMethodModal with TS maybe it's not a bad idea to refactor that to TS too.

Reply:
I will be working on that card next sprint. I guess I can pick it as an adhoc for this sprint instead and combine it in this PR. cc: @farrah-deriv

Reply:
@farhan-nurzi-deriv let's work on it next sprint since the estimation of CancelAddPaymentMethodModal is 2-3days and we are near the end of current sprint. Please add TODO instead.